CONTRIBUTING.md

```diff
@@ -18,22 +18,33 @@ hesitate to ask in the [gitter channel](https://gitter.im/scalacenter/scalafix).
 ## IntelliJ import
 
 The project should import normally into IntelliJ and there should not be any
-false red squiggles. To use the debugger or run tests from withing IntelliJ, run
-at least once `sbt unit/test` to generate a `BuildInfo` file and property files
-for Scalafix testkit.
+false red squiggles. To use the debugger or run tests from within IntelliJ, run
+at least once `sbt unit2_13Target2_13/test` to generate a `BuildInfo` file and
+property files for Scalafix testkit.
 
 ## Testing
 
 Start the SBT shell with `$ sbt`. The commands below assume you have a running
 sbt shell.
 
 ```sh
-> unit/test # Fast unit tests for rules, cli, core. Contains a lot
-            # of different test suites, so it's recommended to use testOnly.
-> unit/testOnly *RuleSuite # Only run tests for rules, using scalafix-testkit.
-> unit/testOnly *RuleSuite -- -z ProcedureSyntax # Only run only ProcedureSyntax unit test.
+# Fast unit tests for rules, cli, core. Contains a lot
+# of different test suites, so it's recommended to use testOnly.
+> unit2_13Target2_13/test
+
+# Only run tests for rules, using scalafix-testkit.
+> unit2_13Target2_13/testOnly *RuleSuite
+
+# Only run only ProcedureSyntax unit test.
+> unit2_13Target2_13/testOnly *RuleSuite -- -z ProcedureSyntax
 ```
 
+[sbt-projectmatrix](https://github.com/sbt/sbt-projectmatrix) is used to
+generate several sbt projects `unitXTargetY` with the same source code,
+but for a combination of Scala versions:
+- used for compiling the framework and rules (`X`)
+- used for compiling and generating SemanticDB files for the test input (`Y`)
```

Comment:
This is not ideal given the tests currently located in `unit`: most of them are NOT related to the target, so we end up running twice these "truely unit" tests.

This is out of scope of this PR, but I believe tests should not remain in a single project.
- Many of them can distributed in existing projects, colocated with the SUT
- Integration tests between cli/reflect/core, etc could be in a module/configuration
- Integration tests depending on shared & testInputs could be in another one